sdk/identity/identity/src/client/identityClient.ts

```diff
@@ -192,16 +216,63 @@ export class IdentityClient extends ServiceClient implements INetworkModule {
     }
   }
 
+  // Here is a custom layer that allows us to abort requests that go through MSAL,
+  // since MSAL doesn't allow us to pass options all the way through.
+
+  generateAbortSignal(correlationId?: string): AbortSignalLike {
+    const controller = new AbortController();
+    const key = correlationId || noCorrelationId;
+
+    const controllers = this.abortControllers.get(key) || [];
+    controllers.push(controller);
+    this.abortControllers.set(key, controllers);
+
+    return controller.signal;
+  }
+
+  abortRequests(correlationId: string = noCorrelationId): void {
+    const key = correlationId || noCorrelationId;
+    const controllers = [
+      ...(this.abortControllers.get(key) || []),
+      // MSAL passes no correlation ID to the get requests...
+      ...(this.abortControllers.get(noCorrelationId) || [])
+    ];
+    if (!controllers.length) {
+      return;
+    }
+    for (const controller of controllers) {
+      controller.abort();
+    }
+    this.abortControllers.set(key, undefined);
+    this.abortControllers.set(noCorrelationId, undefined);
+  }
+
+  getCorrelationId(options?: NetworkRequestOptions): string | undefined {
+    const parameter = options?.body
+      ?.split("&")
+      .map((part: string) => part.split("="))
+      .find(([key]: string[]) => key === "client-request-id");
+    return parameter && parameter.length ? parameter[1] : noCorrelationId;
+  }
+
+  // The MSAL network module methods follow
+
   sendGetRequestAsync<T>(
     url: string,
     options?: NetworkRequestOptions
   ): Promise<NetworkResponse<T>> {
-    const webResource = new WebResource(url, "GET", options?.body, {}, options?.headers);
+    const request = createPipelineRequest({
+      url,
+      method: "GET",
+      body: options?.body,
+      headers: createHttpHeaders(options?.headers),
+      abortSignal: this.generateAbortSignal()
+    });
 
-    return this.sendRequest(webResource).then((response) => {
+    return this.sendRequest(request).then((response) => {
       return {
-        body: response.parsedBody as T,
-        headers: response.headers.rawHeaders(),
+        body: response.bodyAsText ? (parse(response.bodyAsText) as any) : {},
```

Comment:
Just have `parse` be generic so you can pass in the type (or have it return `any` instead of casting here)